test/integration/crud-entity/offline-crud.test.js

```diff
@@ -0,0 +1,179 @@
+
+  const dataStoreTypes = [Kinvey.DataStoreType.Cache, Kinvey.DataStoreType.Sync];
+  const notFoundErrorName = 'NotFoundError';
+  const shouldNotBeCalledErrorMessage = 'Should not be called';
+
+  dataStoreTypes.forEach((currentDataStoreType) => {
+    describe(`${currentDataStoreType} Store CRUD Specific Tests`, () => {
+
+      let networkStore;
+      let syncStore;
+      let cacheStore;
+      let storeToTest;
+      const dataStoreType = currentDataStoreType;
+      const entity1 = common.getSingleEntity(common.randomString());
+      let createdUserIds = [];
+
+      before((done) => {
+        common.cleanUpAppData(collectionName, createdUserIds)
+          .then(() => {
+            return Kinvey.User.signup()
+          })
+          .then((user) => {
+            createdUserIds.push(user.data._id);
+            //store for setup
+            networkStore = Kinvey.DataStore.collection(collectionName, Kinvey.DataStoreType.Network);
+            syncStore = Kinvey.DataStore.collection(collectionName, Kinvey.DataStoreType.Sync);
+            cacheStore = Kinvey.DataStore.collection(collectionName, Kinvey.DataStoreType.Cache);
+            //store to test
+            storeToTest = Kinvey.DataStore.collection(collectionName, dataStoreType);
+            done();
+          })
+          .catch(done)
+      });
+
+      beforeEach((done) => {
+        common.cleanUpCollectionData(collectionName)
+          .then(() => {
+            return cacheStore.save(entity1)
+          })
+          .then(() => done())
+          .catch(done)
+      });
+
+      after((done) => {
+        common.cleanUpAppData(collectionName, createdUserIds)
+          .then(() => done())
+          .catch(done)
+      });
+
+      if (dataStoreType === Kinvey.DataStoreType.Cache) {
+        describe('local cache removal', () => {
+
+          it('find() should remove entities that no longer exist in the backend from the cache', (done) => {
+            const entity = common.getSingleEntity(common.randomString());
+            return storeToTest.save(entity)
+              .then((entity) => {
+                return networkStore.removeById(entity._id)
+              })
+              .then(() => {
+                return storeToTest.find().toPromise()
+              })
+              .then(() => {
+                return syncStore.findById(entity._id).toPromise()
+              })
+              .then(() => {
+                return done(new Error(shouldNotBeCalledErrorMessage));
+              })
+              .catch((error) => {
+                expect(error.name).to.equal(notFoundErrorName);
+                return syncStore.count().toPromise()
+                  .then((count) => {
+                    expect(count).to.equal(1);
+                    done();
+                  })
+              })
+              .catch(done);
+          });
+
+          it.skip('findById() should remove entities that no longer exist on the backend from the cache', (done) => {
```

Comment:
Is this intentionally skipped?

Reply:
Yes, it seems that there is a bug in the covered test case, but should be confirmed, so the tests is skipped for now.